src/diffusers/pipelines/stable_cascade/pipeline_stable_cascade_prior.py

```diff
@@ -0,0 +1,629 @@
+        latent_width = ceil(width / self.config.resolution_multiple)
+        effnet_features_shape = (
+            num_images_per_prompt * batch_size,
+            self.prior.config.c_in,
+            latent_height,
+            latent_width,
+        )
+
+        # 4. Prepare and set timesteps
+        if timesteps is not None:
+            self.scheduler.set_timesteps(timesteps=timesteps, device=device)
+            timesteps = self.scheduler.timesteps
+            num_inference_steps = len(timesteps)
+        else:
+            self.scheduler.set_timesteps(num_inference_steps, device=device)
+            timesteps = self.scheduler.timesteps
+
+        # 5. Prepare latents
+        latents = self.prepare_latents(effnet_features_shape, dtype, device, generator, latents, self.scheduler)
+
+        if isinstance(self.scheduler, DDPMWuerstchenScheduler):
+            timesteps = timesteps[:-1]
+        else:
+            self.scheduler.config.clip_sample = False  # disample sample clipping
+        # 6. Run denoising loop
+        if hasattr(self.scheduler, "betas"):
+            alphas = 1.0 - self.scheduler.betas
+            alphas_cumprod = torch.cumprod(alphas, dim=0)
+        else:
+            alphas_cumprod = []
+
+        self._num_timesteps = len(timesteps)
+        for i, t in enumerate(self.progress_bar(timesteps)):
+            if not isinstance(self.scheduler, DDPMWuerstchenScheduler):
+                if len(alphas_cumprod) > 0:
+                    ratio = self.get_t_condioning(t.long().cpu(), alphas_cumprod)
+                    ratio = ratio.expand(latents.size(0)).to(dtype).to(device)
+                else:
+                    ratio = t.float().div(self.scheduler.timesteps[-1]).expand(latents.size(0)).to(dtype)
+            else:
+                ratio = t.expand(latents.size(0)).to(dtype)
+            # 7. Denoise image embeddings
+            predicted_image_embedding = self.prior(
+                x=torch.cat([latents] * 2) if self.do_classifier_free_guidance else latents,
+                r=torch.cat([ratio] * 2) if self.do_classifier_free_guidance else ratio,
+                clip_text_pooled=text_encoder_pooled,
+                clip_text=text_encoder_hidden_states,
+                clip_img=image_embeds,
+            )
+
+            # 8. Check for classifier free guidance and apply it
+            if self.do_classifier_free_guidance:
+                predicted_image_embedding_text, predicted_image_embedding_uncond = predicted_image_embedding.chunk(2)
+                predicted_image_embedding = torch.lerp(
+                    predicted_image_embedding_uncond, predicted_image_embedding_text, self.guidance_scale
+                )
+
+            # 9. Renoise latents to next timestep
+            if not isinstance(self.scheduler, DDPMWuerstchenScheduler):
+                ratio = t
+            latents = self.scheduler.step(
+                model_output=predicted_image_embedding,
+                timestep=ratio,
```

Comment:
```suggestion
                timestep=t,
```
I think we can just pass `t` here, no?

because: 
if it is not DDPMWuerstchenScheduler,  we want to pass `t`
if it is DDPMWuerstchenScheduler, `t` is same as `ratio`